src/main/java/core/basesyntax/service/functionalityexpansion/FunctionalityExpansion.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.service.functionalityexpansion;
+
+import core.basesyntax.db.CurrentData;
+import core.basesyntax.service.strategy.ShopActivityStrategy;
+import core.basesyntax.service.strategy.ShopBalanceStrategy;
+import core.basesyntax.service.strategy.ShopPurchaseStrategy;
+import core.basesyntax.service.strategy.ShopReturnStrategy;
+import core.basesyntax.service.strategy.ShopSupplyStrategy;
+import java.util.Map;
+
+public class FunctionalityExpansion {
```

Comment:
I suggest to create field (Map<ActivityTypeEnum, ShopActivityStrategy> strategyMap in this class and populate it in constructor for example and here will be method: OperationHandler getHandler(ActivityTypeEnum type)